client/src/components/NearestHospital/HealthFacilities.js

```diff
@@ -151,28 +154,70 @@ const NearestHealthFacilities = props => {
 
   const onSearchInputChange = (value) => {
     setSearchValue(value);
-    console.log(searchValue);
-
   }
   const clearSearch = () => {
     setSearchValue('');
   }
 
-  const [userCoords, setCoords] = useState(null);
+  const [ userCoords, setCoords ] = useState(null);
+  const [ gettingLocation, setGettingLocation ] = useState(false);
 
   const userLocation = () => {
     if(navigator.geolocation) {
-      navigator.geolocation.getCurrentPosition((position, error) => {
-        if(error) {
-          // change error message
-          alert("Sorry we could not get your location. Your browser may not be compatible")
+      setGettingLocation(true);
+
+        const errorCallback_highAccuracy = error => {
+            if (error.code === error.TIMEOUT) {
+              navigator.geolocation.getCurrentPosition(
+                 successCallback,
+                 errorCallback_lowAccuracy,
+                 {maximumAge:600000, timeout:30000, enableHighAccuracy: false}
+               );
+
+              return;
+            }
+            setGettingLocation(false);
+              let errorMsg;
+              if (error.code === 1) {
+                  errorMsg = "PERMISSION_DENIED.";
+                }
+              else if (error.code === 2) {
+                  errorMsg = "POSITION_UNAVAILABLE.";
+                }
+              let msg = `${errorMsg} We could not get your location`;
+              alert(msg);
         }
-        setCoords(position.coords)
 
-      });
+      const errorCallback_lowAccuracy = error => {
+        setGettingLocation(false);
+        let errorMsg;
+        if (error.code === 1) {
+            errorMsg = "PERMISSION_DENIED.";
+          }
+        else if (error.code === 2) {
+            errorMsg = "POSITION_UNAVAILABLE.";
+          }
+        else if (error.code === 3)
+          errorMsg = "TIMEOUT.";
+        let msg = `${errorMsg} We could not get your location`;
+        alert(msg);
+      }
+
+    const successCallback = position => {
+      setGettingLocation(false);
+      setCoords(position.coords)
+    }
+
+    navigator.geolocation.getCurrentPosition(
+      successCallback,
+      errorCallback_highAccuracy,
+      {maximumAge:600000, timeout:25000, enableHighAccuracy: true}
```

Comment:
might be better to extract these values to constants instead of using "magic numbers"